### PR TITLE
Repro for FP S2589: Assignment for captures are ignored

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Repro for #9204
Repro for https://github.com/SonarSource/sonar-dotnet/issues/8885

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs [hunk 1]
```diff
@@ -4223,3 +4223,54 @@ static void IntOverflow_Unchecked_Expression()
...
+        var found = false;
+        Assign();
+        if (!found) // Noncompliant FP
+        {
+            Console.WriteLine("No License for artifact type");
+        }
+        return found;
+
+        void Assign() => found = true; // Assignment in local function
+    }
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs:4275]: This is a duplicate of #8885. But #8885 doesn't seem to have a reproducer. So I guess it's good that we keep it. But link it via comment to the the ticket please.

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs:4277]: I added a reproducer for #8885 as well.

File: analyzers/tests/SonarAnalyzer.Test/Rules/SymbolicExecution/ConditionEvaluatesToConstantTest.cs [hunk 2]
```diff
@@ -57,7 +57,9 @@ public class ConditionEvaluatesToConstantTest
     [DataRow(ProjectType.Test)]
     public void ConditionEvaluatesToConstant_Roslyn_CS(ProjectType projectType) =>
         roslynCS.AddPaths("ConditionEvaluatesToConstant.cs")
-            .AddReferences(NuGetMetadataReference.MicrosoftExtensionsPrimitives("3.1.7").Concat(TestHelper.ProjectTypeReference(projectType)))
+            .AddReferences(NuGetMetadataReference.MicrosoftExtensionsPrimitives("3.1.7"))
+            .AddReferences(MetadataReferenceFacade.SystemComponentModelTypeConverter)
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/Rules/SymbolicExecution/ConditionEvaluatesToConstantTest.cs:61]: For `System.Timers.Timer`

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM

**Pull request merged**